src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2666,10 +2666,16 @@ private bool HasImplicitFunctionTypeConversion(FunctionTypeSymbol source, TypeSy
         {
             if (destination is FunctionTypeSymbol destinationFunctionType)
             {
-                return HasImplicitFunctionTypeToFunctionTypeConversion(source, destinationFunctionType, ref useSiteInfo);
+                if (!HasImplicitFunctionTypeToFunctionTypeConversion(source, destinationFunctionType, ref useSiteInfo))
+                {
+                    return false;
+                }
             }
-
-            return IsValidFunctionTypeConversionTarget(destination, ref useSiteInfo);
+            else if (!IsValidFunctionTypeConversionTarget(destination, ref useSiteInfo))
+            {
+                return false;
+            }
+            return source.GetInternalDelegateType() is { };
```

Comment:
>return source.GetInternalDelegateType() is { }; [](http://example.com/codeflow?start=12&length=47)

How confident are we that all places like this one got fixed up? I mean places that do not explicitly use ```GetInternalDelegateType``` API, but assume that it is going to return a non-null value just because we are dealing with ```FunctionTypeSymbol``` and before this PR that meant that the signature was inferred successfully.

Reply:
I believe I have reviewed all callers that use `FunctionTypeSymbol` directly or indirectly.

The most interesting cases are in `MethodTypeInferrer` where the bounds collection was (and still is) including `FunctionTypeSymbol` instances in the set of lower bounds and which now explicitly drops bounds with `null` delegate type in `MethodTypeInferrer.Fix()`.